src/k8s/pkg/component/component.go

```diff
@@ -14,8 +14,9 @@ import (
 
 // ComponentManager defines an interface for managing k8s components.
 type ComponentManager interface {
-	// Enable enables a k8s component.
+	// Enable enables a k8s component, optionally specifying custom configuration options.
 	Enable(name string) error
+	EnableWithValues(name string, values map[string]any) error
```

Comment:
if `Enable(name)` is equivalent to `EnableWithValues(name, nil)`, then let's only have a single one and update all callers:

```go
Enable(name string, values map[string]any) error
```